### ISSUE TITLE
Execution failed for running gradle project

### ISSUE THREAD

**Issue by leawang0** [nus-cs2103-AY2324S1/forum#47]
I am running my project in WSL Ubuntu. After I converted the normal java project to the gradle project, I am unable to run it even after modifying build.gradle. 

The error message is:
```text
Execution failed for task ':run'.
> Process 'command '/usr/lib/jvm/java-11-openjdk-amd64/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':run'. (31 internal lines...)
	Caused by: org.gradle.process.internal.ExecException: Process 'command
'/usr/lib/jvm/java-11-openjdk-amd64/bin/java'' finished with non-zero exit value 1(119 internal lines...)
```

I have tried using stack overflow but yield no results. Someone mentioned that making sure the java jdk used by Gradle is the same as the project may help, but I tried and did not work.

**Screenshot of my project structure and build.gradle file:**
![image](https://github.com/nus-cs2103-AY2324S1/forum/assets/83194176/b64e2595-b800-489d-b8c4-3a312f9e6f93)

**Screenshot of the error message:
When I try to debug with more flags like '--info' or '--debug', the error message is basically the same as this one, there is no additional information given by gradle.**
![image](https://github.com/nus-cs2103-AY2324S1/forum/assets/83194176/af1c5852-bf78-4942-b1e9-214248a7f00f)

**Screenshot of Java SDK configuration:**
![image](https://github.com/nus-cs2103-AY2324S1/forum/assets/83194176/8e557857-00ff-4f04-a006-145855ca4b3e)
![image](https://github.com/nus-cs2103-AY2324S1/forum/assets/83194176/a660d7e5-202c-44dc-8e38-bd1c8aced08c)

**Comment by damithc** [comment]
@wanghejin somewhat related: does your version of WSL support GUI applications? If not, you might run into issues later when we add a GUI to the iP.

**Comment by leawang0** [comment]
I am not exactly sure...I'm running Ubuntu 22.04😢 I did encounter difficulties launching GUI for apache jmeter before (broken GUI that is partially shown) and I couldn't solve it. Not sure if this time the same problem will occur. 

May I know if WSL is related to the problem of running gradle project? Also, what do I do if I indeed run into issues adding GUI later...

**Comment by damithc** [comment]
> Also, what do I do if I indeed run into issues adding GUI later...

One option is to use a Windows terminal (e.g. MS-DOS terminal) instead of WSL. That way, both development and testing will be done on the same OS.

**Comment by KamJiaYue** [comment]
Try this:

build.gradle
applications
mainClass.set(''tasks.Cheems'')

**Comment by leawang0** [comment]
Hi I tried this but did not work. Setting mainClass to tasks.Cheems gives me `Could not find or load main class java.Cheems` error.
Only setting it to `Cheems` will not have this error.

**Comment by damithc** [comment]
> Hi I tried this but did not work. Setting mainClass to tasks.Cheems gives me `Could not find or load main class java.Cheems` error.

It is neither `tasks.Cheems` nor `java.Cheems`, but should be `cheems.Cheems` (as per the location of this class https://github.com/wanghejin/ip/blob/master/src/main/java/cheems/Cheems.java)

**Comment by leawang0** [comment]
Yes, I added the package in recent commits. However, the same error still persists.

**Comment by Vanessamae23** [comment]
what is the header of your Cheems?
maybe if u put package tasks in Cheems then do tasks.Cheems in the mainClass.set()

**Comment by leawang0** [comment]
Thank you for those who took the time to help me debug. I have solved the problem by running git bash instead of Ubuntu. I believe it is a Ubuntu problem. I tried switching to powershell / command prompt first but their commands are different from the Linux commands I am used to. Git bash provides a command line interface for the windows system, emulating the linux environment. Hence simpler for me. Hope there won't be any more problems.
